resources/analysisTools/snvPipeline/in_dbSNPcounter.pl

```diff
@@ -37,85 +37,80 @@
 
 
 my $header = "";
-while ($header = <FH>)
-{
-	last if ($header =~ /^\#CHROM/); # that is the line with the column names
-}
-chomp $header;
-
-my $DBSBP = 13;
-my $KG = 14;
-my $CONFIDENCE = 28;
-my $CANNOTATION = -1;
-my $RECLASSIFICATION = -1;
-my $CLASSIFICATION;
+my ($DBSBP, $KG, $CONFIDENCE, $CLASSIFICATION);
 
-@help = split(/\t/, $header);
-for (my $c = 0; $c < @help; $c++)
-{
-	# fixed column labels (from vcf_pileup_compare)
-	if ($help[$c] eq "CONFIDENCE")
-	{
-		$CONFIDENCE = $c;
-		print STDERR "CONFIDENCE in column $c\n";
-	}
-	if ($help[$c] eq "1K_GENOMES")
+while(!eof($FH)){
+	my $line = readline($FH) || die "Could not read line: $?";
+	chomp $line;
+	if ($line =~ /^#/)
 	{
-		$KG = $c;
-		print STDERR "INFO in column $c\n";
-	}
-	if ($help[$c] eq "DBSNP")
-	{
-		$DBSBP = $c;
-		print STDERR "DBSNP_COL in column $c\n";
-	}
-	if ($help[$c] eq "ANNOTATION_control")
-    {
-    	$CANNOTATION = $c;
-    	print STDERR "ANNOTATION_control in column $c\n";
-    }
-    if ($help[$c] eq "RECLASSIFICATION")
-    {
-        $RECLASSIFICATION = $c;
-        print STDERR "RECLASSIFICATION in column $c\n";
-    }
-}
+		#print STDOUT "HELLO\n";
+		if ($line =~/^#CHROM/){
+			chomp $line;
 
-if ($CANNOTATION > 0) {
-    $CLASSIFICATION = $CANNOTATION;
-} else {
-    $CLASSIFICATION = $RECLASSIFICATION;
-}
+			my @head=split("\t", $line);
+			my %col;
 
-# I know that it's evilly hardcoded!
-while (<FH>)
-{
-	if ($_ =~ /^#/)	# skip header lines
-	{
-		next;
-	}
-	$all++;
-	@help = split ("\t", $_);
-	if ($help[$CLASSIFICATION] =~ /somatic/)
-	{
-		$somatic++;
-		if ($help[$CONFIDENCE] >= $minscore)
-		{
-			$scored++;
-			#print STDERR $_;
-			if ($help[$DBSBP] =~ /MATCH=exact/)
+			my $i = 0;
+			while($i < @head)
 			{
-				$dbSNP++;
+				if($head[$i] eq "DBSNP"){
+					$col{"DBSNP"} = $i;
+					print STDERR "DBSNP in column ", $i+1,"\n";
+				}
+				if($head[$i] eq "1K_GENOMES"){
+					$col{"1K_GENOMES"} = $i;
+					print STDERR "1K_GENOMES in column ", $i+1,"\n";
+				}
+				if($head[$i] eq "RECLASSIFICATION"){
+					$col{"RECLASSIFICATION"} = $i;
+					print STDERR "RECLASSIFICATION in column ", $i+1,"\n";
+				}
+				if($head[$i] eq "ANNOTATION_control"){
+					$col{"ANNOTATION_control"} = $i;
+					print STDERR "ANNOTATION_control in column ", $i+1,"\n";
+				}
+				if($head[$i] eq "CONFIDENCE"){
+					$col{"CONFIDENCE"} = $i;
+					print STDERR "CONFIDENCE in column ", $i+1,"\n";
+				}
+				$i++;
 			}
-			if ($help[$KG] =~ /MATCH=exact/)
+
+			$DBSBP = $col{"DBSNP"};
+			$KG = $col{"1K_GENOMES"};
+			$CONFIDENCE = $col{"CONFIDENCE"};
+
+			if ($col{"ANNOTATION_control"} > 0) {
+				$CLASSIFICATION = $col{"ANNOTATION_control"};
+			} else {
+				$CLASSIFICATION = $col{"RECLASSIFICATION"};
+			}
+		}
+	}
+	else{
+		$all++;
+		@help = split ("\t", $line);
```

Comment:
You could also rename this to `@line`. I think, it also has advantages if two variables that differ only by the type, but not by the content (in principle) are called the same. Some for `@head` above.